crates/relayer-cli/src/commands/evidence.rs

```diff
@@ -0,0 +1,651 @@
+    rt: Arc<TokioRuntime>,
+    chain: &CosmosSdkChain,
+    key_name: Option<&String>,
+    dv: DuplicateVoteEvidence,
+) -> eyre::Result<()> {
+    let config = app_config();
+
+    // Fetch all the counterparty clients of this chain.
+    let counterparty_clients = fetch_all_counterparty_clients(chain)?;
+
+    let mut chains = HashMap::new();
+
+    // For each counterparty client, build the double voting evidence and submit it to the chain,
+    // freezing that client.
+    for (counterparty_chain_id, _counterparty_client_id) in counterparty_clients {
+        if !chains.contains_key(&counterparty_chain_id) {
+            let chain_handle = spawn_chain_runtime_with_modified_config::<BaseChainHandle>(
+                &config,
+                &counterparty_chain_id,
+                rt.clone(),
+                |chain_config| {
+                    if let Some(key_name) = key_name {
+                        chain_config.key_name = key_name.to_string();
+                    }
+                },
+            )?;
+
+            chains.insert(counterparty_chain_id.clone(), chain_handle);
+        };
+
+        let counterparty_chain_handle = chains.get(&counterparty_chain_id).unwrap();
+
+        let signer = counterparty_chain_handle.get_signer()?;
+
+        // If the misbehaving chain is a CCV consumer chain,
+        // then try fetch the consumer chains of the counterparty chains.
+        // If that fails, then that chain is not a provider chain.
+        // Otherwise, check if the misbehaving chain is a consumer of the counterparty chain,
+        // which is definitely a provider.
+        let counterparty_is_provider = if chain.config().ccv_consumer_chain {
+            let consumer_chains = counterparty_chain_handle
+                .query_consumer_chains()
+                .unwrap_or_default(); // If the query fails, use an empty list of consumers
+
+            // FIXME: Do do we need to check if the client id also matches?
+            // ie. is it okay to submit a `MsgSubmitIcsConsumerDoubleVoting` to all clients
+            // of the provider chain, or should we only do this for the CCV client, and
+            // use the standard message for other clients?
+            consumer_chains
+                .iter()
+                .any(|(chain_id, _)| chain_id == chain.id())
+        } else {
+            false
+        };
+
+        if !counterparty_is_provider {
+            debug!("counterparty chain {counterparty_chain_id} is not a CCV provider chain, skipping...");
+            continue;
+        }
+
+        info!("submitting CCV misbehaviour to provider chain {counterparty_chain_id}");
+
+        // Construct the light client block header for the consumer chain at the infraction height
+        let infraction_block_header = {
+            let infraction_height = dv.vote_a.height;
+
+            let trusted_validator_set = rt
+                .block_on(chain.rpc_client.validators(infraction_height, Paging::All))?
```

Comment:
As in the comment below; this is wrong should use the `trusted_height` instead of the` infraction_height`.